indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunner.java

```diff
@@ -1717,6 +1717,98 @@ public Response getUnparseableEvents(
     return Response.ok(parseExceptionHandler.getSavedParseExceptionReports()).build();
   }
 
+  @POST
+  @Path("/updateConfig")
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response updateConfig(TaskConfigUpdateRequest<PartitionIdType, SequenceOffsetType> req) throws InterruptedException
+  {
+    try {
+      requestPause();
+      checkpointSequences();
+
+      this.ioConfig = req.getIoConfig();
+      this.stream = ioConfig.getStartSequenceNumbers().getStream();
+      this.endOffsets = new ConcurrentHashMap<>(ioConfig.getEndSequenceNumbers().getPartitionSequenceNumberMap());
+      minMessageTime = Configs.valueOrDefault(ioConfig.getMinimumMessageTime(), DateTimes.MIN);
+      maxMessageTime = Configs.valueOrDefault(ioConfig.getMaximumMessageTime(), DateTimes.MAX);
+
+      createNewSequenceFromIoConfig(req.getIoConfig());
+      resume();
+      return Response.ok().build();
+    } catch (Exception e) {
+      log.makeAlert(e, "Failed to update task config");
+      return Response.serverError().entity(e.getMessage()).build();
+    }
+  }
+
+  /**
+   * Creates new sequences for the ingestion process. It currently accepts the ioConfig given by the request as the correct offsets
+   * and ignores the offsets it may have stored in currOffsets and endOffsets.
+   */
+  private void createNewSequenceFromIoConfig(SeekableStreamIndexTaskIOConfig<PartitionIdType, SequenceOffsetType> ioConfig)
```

Comment:
For the 2 new methods, the `SeekableStreamIndexTaskRunner` must already be performing these actions.
Let's try to put them in common methods so that we can use the same method in the normal flow as well as on update config.